Catch a moment: some tests fail for Yarik

#### What is the problem?
```shell
$> WRAPT_DISABLE_EXTENSIONS=1 DATALAD_TESTS_SSH=1 python -m nose -s -v datalad/interface/tests/test_utils.py:test_save_hierarchy datalad/distribution/tests/test_clone.py:test_failed_clone datalad/distribution/tests/test_install.py:test_failed_install datalad/metadata/tests/test_base.py:test_aggregation
datalad.interface.tests.test_utils.test_save_hierarchy ... ERROR
datalad.distribution.tests.test_clone.test_failed_clone ... FAIL
datalad.distribution.tests.test_install.test_failed_install ... FAIL
datalad.metadata.tests.test_base.test_aggregation ... FAIL
Versions: appdirs=1.4.0 boto=2.44.0 cmd:annex=6.20170815+gitg22da64d0f-1~ndall+1 cmd:git=2.11.0 cmd:system-git=2.14.1 cmd:system-ssh=7.4p1 git=2.1.6 gitdb=2.0.0 humanize=0.5.1 iso8601=0.1.11 msgpack=0.4.8 requests=2.12.4 scrapy=1.4.0 six=1.10.0 tqdm=4.11.2 wrapt=1.10.11

======================================================================
ERROR: datalad.interface.tests.test_utils.test_save_hierarchy
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/yoh/proj/datalad/datalad/datalad/tests/utils.py", line 458, in newfunc
    return t(*(arg + (d,)), **kw)
  File "/home/yoh/proj/datalad/datalad/datalad/interface/tests/test_utils.py", line 194, in test_save_hierarchy
    super_datasets=True)
  File "/home/yoh/proj/datalad/datalad/venvs/dev/local/lib/python2.7/site-packages/wrapt/wrappers.py", line 562, in __call__
    args, kwargs)
  File "/home/yoh/proj/datalad/datalad/datalad/distribution/dataset.py", line 437, in apply_func
    return f(**kwargs)
  File "/home/yoh/proj/datalad/datalad/venvs/dev/local/lib/python2.7/site-packages/wrapt/wrappers.py", line 523, in __call__
    args, kwargs)
  File "/home/yoh/proj/datalad/datalad/datalad/interface/utils.py", line 437, in eval_func
    return return_func(generator_func)(*args, **kwargs)
  File "/home/yoh/proj/datalad/datalad/venvs/dev/local/lib/python2.7/site-packages/wrapt/wrappers.py", line 523, in __call__
    args, kwargs)
  File "/home/yoh/proj/datalad/datalad/datalad/interface/utils.py", line 425, in return_func
    results = list(results)
  File "/home/yoh/proj/datalad/datalad/datalad/interface/utils.py", line 382, in generator_func
    result_renderer, result_xfm, _result_filter, **_kwargs):
  File "/home/yoh/proj/datalad/datalad/datalad/interface/utils.py", line 449, in _process_results
    for res in results:
  File "/home/yoh/proj/datalad/datalad/datalad/interface/save.py", line 367, in __call__
    message=message)
  File "/home/yoh/proj/datalad/datalad/datalad/interface/save.py", line 112, in save_dataset
    update=save_entire_ds)
  File "/home/yoh/proj/datalad/datalad/datalad/support/gitrepo.py", line 270, in newfunc
    files_new = [normalize(self.path, path) for path in files]
  File "/home/yoh/proj/datalad/datalad/datalad/support/gitrepo.py", line 190, in _normalize_path
    % path, filename=path)
FileNotInRepositoryError: FileNotInRepositoryError: 
Path outside repository: /tmp/datalad_temp_tree_test_save_hierarchyES2q4T
[Errno None] : Path outside repository: /tmp/datalad_temp_tree_test_save_hierarchyES2q4T: '/tmp/datalad_temp_tree_test_save_hierarchyES2q4T'
-------------------- >> begin captured logging << --------------------
datalad.utils: Level 5: Importing datalad.utils
datalad.utils: Level 5: Done importing datalad.utils
datalad.cmd: Level 9: Will use git under '/usr/lib/git-annex.linux' (no adjustments to PATH if empty string)
datalad.cmd: Level 9: Running: ['git', 'version']
datalad.cmd: Level 8: Finished running ['git', 'version'] with status 0
datalad.cmd: Level 9: Running: ['git', 'config', '-z', '-l', '--show-origin']
datalad.cmd: Level 8: Finished running ['git', 'config', '-z', '-l', '--show-origin'] with status 0
datalad.ui: Level 5: Starting importing ui
datalad.ui.dialog: Level 5: Starting importing ui.dialog
datalad.ui.dialog: Level 5: Done importing ui.dialog
datalad.ui: Level 5: Initiating UI switcher
datalad.ui: DEBUG: UI set to DialogUI(out=<file>)
datalad.ui: Level 5: Done importing ui
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: datalad.distribution.tests.test_clone.test_failed_clone
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/yoh/proj/datalad/datalad/datalad/tests/utils.py", line 593, in newfunc
    return t(*(arg + (filename,)), **kw)
  File "/home/yoh/proj/datalad/datalad/datalad/distribution/tests/test_clone.py", line 286, in test_failed_clone
    assert_status('error', res)
  File "/home/yoh/proj/datalad/datalad/datalad/tests/utils.py", line 1120, in assert_status
    dumps(results, indent=1, default=lambda x: "<not serializable>")))
AssertionError: Test 1/1: expected status ['error'] not found in:
[
 {
  "status": "ok", 
  "source_url": "http://nonexistingreallyanything.somewhere/bla", 
  "action": "install", 
  "path": "/tmp/datalad_temp_test_failed_cloneIFFnNw/sub", 
  "type": "dataset", 
  "refds": "/tmp/datalad_temp_test_failed_cloneIFFnNw"
 }
]

======================================================================
FAIL: datalad.distribution.tests.test_install.test_failed_install
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/yoh/proj/datalad/datalad/datalad/tests/utils.py", line 593, in newfunc
    return t(*(arg + (filename,)), **kw)
  File "/home/yoh/proj/datalad/datalad/datalad/distribution/tests/test_install.py", line 582, in test_failed_install
    source="http://nonexistingreallyanything.somewhere/bla")
AssertionError: IncompleteResultsError not raised

======================================================================
FAIL: datalad.metadata.tests.test_base.test_aggregation
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/yoh/proj/datalad/datalad/datalad/tests/utils.py", line 806, in newfunc
    return func(*args, **kwargs)
  File "/home/yoh/proj/datalad/datalad/datalad/tests/utils.py", line 458, in newfunc
    return t(*(arg + (d,)), **kw)
  File "/home/yoh/proj/datalad/datalad/datalad/metadata/tests/test_base.py", line 143, in test_aggregation
    assert_raises(InsufficientArgumentsError, aggregate_metadata, None)
AssertionError: InsufficientArgumentsError not raised

----------------------------------------------------------------------
Ran 4 tests in 38.003s

FAILED (errors=1, failures=3)
WRAPT_DISABLE_EXTENSIONS=1 DATALAD_TESTS_SSH=1 python -m nose -s -v      14.67s user 6.75s system 54% cpu 39.360 total
```
Was testing bleeding edge annex, but got the same errors even when downgraded... I guess they need to become a bit more robust (unlikely to be my git/annex configuration issue, but could be)

--- a/datalad/tests/test_base.py
+++ b/datalad/tests/test_base.py
@@ -1,0 +1,34 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+
+from .utils import (
+    chpwd,
+    get_dataset_root,
+    with_tree,
+
+    assert_raises,
+)
+
+
+#@with_tempfile(mkdir=True)
+# with_tempfile dereferences tempdir, so does not trigger the failure
+# on Yarik's laptop where TMPDIR=~/.tmp and ~/.tmp -> /tmp.
+# with_tree in turn just passes that ~/.tmp/ directory
+@with_tree(tree={})
+def test_not_under_git(path):
+    from datalad.distribution.dataset import require_dataset
+    dsroot = get_dataset_root(path)
+    assert dsroot is None, "There must be no dataset above tmp %s. Got: %s" % (path, dsroot)
+    with chpwd(path):
+        # And require_dataset must puke also
+        assert_raises(
+            Exception,
+            require_dataset,
+            None, check_installed=True, purpose='test'
+        )

